src/services/dashboard.js

```diff
@@ -1,22 +1,43 @@
 /* eslint-disable no-param-reassign */
 /* eslint-disable no-unused-vars */
 /* eslint-disable no-underscore-dangle */
-import request from '../utils/request';
+import { notification } from 'antd';
+import request, { codeMessage } from '../utils/request';
 import { getAllMonthsWithinRange } from '../utils/moment_constants';
 
 async function scrollUntilEmpty(datastoreConfig, data) {
   const endpoint = `${datastoreConfig.elasticsearch}/_search/scroll?scroll=1m`;
   const allData = data;
-
   if (allData.hits.total !== allData.hits.hits.length) {
-    const scroll = Promise.resolve(request.post(`${endpoint}&scroll_id=${allData._scroll_id}`));
+    const scroll = Promise.resolve(
+      request.post(`${endpoint}&scroll_id=${allData._scroll_id}`, {
```

Comment:
Ah; I wasn't entirely sure before, but I see that you were getting the 502 on the follow-up scroll queries rather than on the initial query. That makes even less sense to me, and I wish we had more context to figure out what's going on with Elasticsearch here. I'm still concerned that most of our code isn't protected against unexpected Elasticsearch errors; for example, if we got the 502 on the initial request rather than on the scroll queries.

Then again, this is maintenance on the old dashboard code, so limiting the change is a good idea. What can we do on `main` to make our dashboard more robust in handling back-end and connectivity errors?

Reply:
I believe we have error handling work #185 in the `main` branch that catches errors like these and displays them to the user.